CONTRIBUTING.md

```diff
@@ -1,31 +1,117 @@
+
+
+## Major contributions
+
+If your contribution is a major feature or breaking change,
+then you must first write an XRP Ledger Standard (XLS) describing it.
+Go to [XRPL-Standards](https://github.com/XRPLF/XRPL-Standards/discussions),
+choose the next available standard number, and
+open a discussion with an appropriate title to propose your draft standard.
+
+When you submit a PR for this change, please link the corresponding XLS in the
+description.
+
+
+## Before making a pull request
+
+Changes that alter transaction processing must be guarded by an
+[Amendment](https://xrpl.org/amendments.html).
+All other changes that maintain the existing behavior do not need an
+Amendment.
+
+Ensure that your code compiles according to the build instructions in
+[`BUILD.md`](./BUILD.md).
+If you create new source files, they must go under `src/ripple`.
+You will need to add them to one of the
+[source lists](./Builds/CMake/RippledCore.cmake) in CMake.
+
+Please write tests for your code.
+If you create new test source files, they must go under `src/test`.
+You will need to add them to one of the
+[source lists](./Builds/CMake/RippledCore.cmake) in CMake.
+If your test can be run offline, in under 60 seconds, then it can be an
+automatic test run by `rippled --unittest`.
+Otherwise, it must be a manual test.
+
+The source must be formatted according to the style guide below.
+
+Header includes must be [levelized](./Builds/levelization).
+
+
+## Pull requests
+
+Pull requests must target the `develop` branch.[^1]
+
+[^1]: There are exceptions to this policy for hotfixes, but no one consulting
+  this document will be in that situation.
+
+Changes to pull requests must be added as new commits.
+You may **never force push a branch in a pull request** (e.g. after a rebase).
+This preserves the ability for reviewers to filter changes since their last
+review.
+
+A pull request must obtain **approvals from at least two reviewers** before it
+can be considered for merge by a Maintainer.
+Maintainers retain discretion to require more approvals if they feel the
+credibility of the existing approvals is insufficient.
+
+Pull requests must be merged by [squash-and-merge][2]
+to preserve a linear history for the `develop` branch.
+
 
 # Style guide
-This is a non-exhaustive list of recommended style guidelines. These are not always strictly enforced and serve as a way to keep the codebase coherent rather than a set of _thou shalt not_ commandments.
+
+This is a non-exhaustive list of recommended style guidelines.
+These are not always strictly enforced and serve as a way to keep the codebase coherent rather than a set of _thou shalt not_ commandments.
+
 
 ## Formatting
-All code must conform to `clang-format` version 10, unless the result would be unreasonably difficult to read or maintain.
-To change your code to conform use `clang-format -i <your changed files>`.
```

Comment:
This might be worth keeping in, it should be easier to do the right thing than documenting only how to ignore the formatter. ;-)

Reply:
I tjhink that documenting how to use clang-format is outside the scope of this document, but I will add this example back in.